src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((acc: GroupsMap<T>, item: T) => {
+    const groupKey = item[key] as unknown as string;
```

Comment:
Casting `item[key]` directly to `string` might lead to runtime errors if `item[key]` is not actually a string. It would be safer to ensure the key's value is a string or to handle cases where it is not.